### PR TITLE
image,seed/seedwriter: fix corner cases of asserts fetching not triggered

### PR DESCRIPTION
**Pull request by pedronis**
with not extended classic models never seeing a system snap or seeing snapd outside the model would not trigger assertion fetching

### CODE REVIEW

**Review by jawn-smith — APPROVED** [review]

**Comment by jawn-smith** [comment]
I've compiled this version into ubuntu-image and confirmed that it solves my issues. Thanks!

File: seed/seedwriter/seed16.go [hunk 1]
```diff
@@ -171,7 +181,14 @@ func (pol *policy16) isSystemSnapCandidate(sn *SeedSnap) bool {
 			return true
 		}
 	}
-	return pol.model.Classic() && sn.SnapName() == "core"
+	if pol.model.Classic() {
+		if pol.snapdUsedOnClassic {
+			return sn.SnapName() == "snapd"
+		} else {
+			return sn.SnapName() == "core"
```
Comment (Meulengracht) [inline seed/seedwriter/seed16.go:188]: What is the difference in these two scenarios? Do we have classic images without snapd?

File: seed/seedwriter/writer.go [hunk 2]
```diff
@@ -876,6 +880,21 @@ func (w *Writer) checkBase(info *snap.Info, modes []string) error {
...
+	}
+
+	for _, optSnap := range w.optionsSnaps {
+		snapName := optSnap.Name
+		sn := w.localSnaps[optSnap]
+		if sn != nil {
+			snapName = sn.Info.SnapName()
+		}
+		w.policy.recordSnapNameUsage(snapName)
+	}
```
Comment (Meulengracht) [inline seed/seedwriter/writer.go:895]: this part of the function was a bit unexpected from the usage, shouldn't this be done somewhere else and only once?

Comment (Meulengracht) [reply seed/seedwriter/writer.go:895]: Ah right - the model step is only once of course

**Pull request merged**